File: julia/p1.jl
function mults()
  mults = [mod(i, 3) == 0 ? i : mod(i,5) == 0 ? i : 0 for i=1:999]
  sum(mults)
end

t1 = time()
i = 0
while i < 10000
  mults()
  i += 1
end
t2 = time()
println("elapsed time: ", t2 - t1, " seconds")


t1 = time()
i = 0
while i < 10000
  m = [mod(i, 3) == 0 ? i : mod(i,5)==0 ? i : 0 for i=1:999]
  sum(m)
  i += 1
end
t2 = time()
println("elapsed time: ", t2 - t1, " seconds")

File: go/p1.go
package main

import (
  "fmt"
  "time"
)

func mults() int {
  sum := 0

  for i := 0; i < 1000; i += 1 {
    if i % 3.0 == 0 || i % 5.0 == 0 {
      sum += i
    }
  }

  return sum
}

func main() {
  s := time.Now()

  for i := 0; i < 10000; i += 1 {
    mults()
  }

  fmt.Println(time.Now().Sub(s))
}
